File: backend/src/main/java/com/project2/domain/member/dto/AuthorDTO.java
package com.project2.domain.member.dto;

import lombok.Getter;

@Getter
public class AuthorDTO {
	private Long memberId;
	private String nickname;
	private String profileImageUrl;

	public AuthorDTO(Long memberId, String nickname, String profileImageUrl) {
		this.memberId = memberId;
		this.nickname = nickname;
		this.profileImageUrl = profileImageUrl;
	}
}

File: backend/src/main/kotlin/com/project2/domain/member/dto/AuthorDTO.kt
package com.project2.domain.member.dto

import com.project2.domain.member.entity.Member

data class AuthorDTO(val memberId: Long, val nickname: String, val profileImageUrl: String) {
    companion object {
        fun from(member: Member): AuthorDTO = AuthorDTO(memberId = member.id!!, nickname = member.nickname, profileImageUrl = member.getProfileImageUrlOrDefaultUrl())
    }
}
